### PR TITLE
VB-1474, Upcoming visits: show BOOKED and CANCELLED visits

### PR DESCRIPTION
**Pull request by hutcheonb-moj**
## Description
Display upcoming cancelled visits on booking search and prisoner upcoming visits pages

Removed timestamp from upcomingVisits call.
Faked time on the apiClient test for upcomingVisits

Rebased from main

### CODE REVIEW

File: server/data/visitSchedulerApiClient.ts [hunk 1]
```diff
@@ -39,13 +39,24 @@ class VisitSchedulerApiClient {
...
+      path: '/visits',
+      query: new URLSearchParams({
+        prisonerId: offenderNo,
+        startTimestamp: startTimestamp || new Date().toISOString(),
+        visitStatus: 'BOOKED,CANCELLED',
+      }).toString(),
+    })
+  }
+
   getUpcomingVisits(offenderNo: string, startTimestamp?: string): Promise<Visit[]> {
```
[the remark this answers is not in the document]
Comment (tpmcgowan) [reply server/data/visitSchedulerApiClient.ts:53]: I actually mean something like `... visitStatuses: Visit['visitStatus'][] = ['BOOKED'], ...` so TypeScript ensures only allowed status get passed.

[the remark this answers is not in the document]
Comment (psoleckimoj) [reply server/data/visitSchedulerApiClient.ts:53]: Agree with @tpmcgowan , it'd mean testing is easier.

File: server/services/visitSessionsService.ts [hunk 2]
```diff
@@ -251,6 +251,24 @@ export default class VisitSessionsService {
...
+    username: string
+    offenderNo: string
+  }): Promise<VisitInformation[]> {
+    const token = await this.systemToken(username)
+    const visitSchedulerApiClient = this.visitSchedulerApiClientBuilder(token)
+
+    logger.info(`Get booked and cancelled upcoming visits for ${offenderNo}`)
+    const visits = await visitSchedulerApiClient.getUpcomingVisitsCancelledAndBooked(offenderNo)
+
+    return visits.map(visit => {
```
Comment (psoleckimoj) [inline server/services/visitSessionsService.ts:267]: Can make this simpler 🙂 

`return visits.map(visit => this.buildVisitInformation(visit))`

**Review by tpmcgowan — APPROVED** [review]
LGTM!

**Pull request merged**